### ISSUE TITLE
Static linked C++ option: ALSOFT_STATIC_STDCXX does not really remove link to libstdc++.so.6 on Linux

### ISSUE THREAD

**Issue by fuzhouch** [kcat/openal-soft#952]
Hi,

   I have a question when trying to build a portable shared libopenal.so for my game project. I noticed the flags used for ALSOFT_STATIC_STDCXX is ``-Wl,--push-state,-Bstatic,-lstdc++,--pop-state``. With this option enabled, the generated executable still dynamically links on libstdc++.so.6.

Following GCC documentation, the static link option for C++ should be ``-static-libstdc++``. I'm wondering why we don't use -static-libstdc++ but use ``-Wl,--push-state,-Bstatic,-lstdc++,--pop-state``.

### The reason I ask this question

As a game developer, an explicit link to libstdc++.so.6 causes a runtime crash when copying my pre-built binaries to players' machines. A typical scenario is: my game is built from latest Ubuntu devbox, then Steam distributs game to players' machines with only Steam runtime.

The root cause of runtime break is because libstdc++.so.6 internally maintain an GLIBCXX_{x}.{y}.{z} version string (visible via ``readelf -a -W``). When the binary is built in a newer OS but running on player machine with older runtime due to version mismatch. With -static-libstdc++, GCC completely embeds C++ runtime as part of shared library, which solves this problem.

To clarify: There are many scenarios which do not need this behavior.  For exmaple, when building an OS-provided package (e.g. .deb or .rpm), we should intentionally use shared libraries to save harddisk. Gaming is less often case, but the use of static link is a common practice in game development.

```
# Simple example when compiling a hello world C++ shared library:
fuzhouch@fcdvh01 ~/p/learning> g++ -shared -fPIC helloworld.cpp -o libhelloworld.so -Wl,--push-state,-Bstatic,-lstdc++,--pop-state
fuzhouch@fcdvh01 ~/p/learning> ls -l libhelloworld.so && ldd libhelloworld.so
-rwxr-xr-x 1 fuzhouch fuzhouch 1868112 Dec 19 17:41 libhelloworld.so*
	linux-vdso.so.1 (0x00007ffeac5fe000)
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f21caa00000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f21ca913000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f21cad3a000)  ### <<<<<< Both libgcc_s.so.1 and libstdc++.so.6 are linked
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f21ca731000)
	/usr/lib64/ld-linux-x86-64.so.2 (0x00007f21caef7000)
fuzhouch@fcdvh01 ~/p/learning> g++ -shared -fPIC helloworld.cpp -o libhelloworld.so -static-libgcc -static-libstdc++
fuzhouch@fcdvh01 ~/p/learning> ls -l libhelloworld.so && ldd libhelloworld.so
-rwxr-xr-x 1 fuzhouch fuzhouch 1905232 Dec 19 17:43 libhelloworld.so*
	linux-vdso.so.1 (0x00007ffc747b1000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007fc893b8f000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007fc8939ad000)     ### <<<<<< Both libgcc_s.so.1 and libstdc++.so.6 are embeded
	/usr/lib64/ld-linux-x86-64.so.2 (0x00007fc893e1d000)

```

**Comment by kcat** [comment]
I actually thought I tried `-static-libstdc++` when I first added the option and it didn't work (either a link error, or it didn't static link with MinGW, I don't remember which). Though trying again now, and it does seem to. Should be fixed with commit 59c466077fd6f16af64afcc6260bb61aa4e632dc.

**Comment by fuzhouch** [comment]
Thanks @kcat for quick response! Yup this commit should work on native Linux.
